POC-OOBE/OOBE-Sandbox/PowerToys Settings Sandbox/Services/NotificationService.cs

```diff
@@ -10,11 +10,98 @@ internal class NotificationService
 
+            ToastVisual visual = createToastVisual(title, content, logo);
+
+            string onActionClick = new QueryString()
+                {
+                    { "action", "openApp"},
+                    { "status", "openFirst"},
+                }.ToString();
+
+            ToastActionsCustom actions = createToastAction("Get Started", onActionClick);
+
+            string onLaunch = new QueryString()
+                {
+                    {"action", "openApp"},
+                }.ToString();
+
+            ToastContent toastContent = createToastContent(visual, actions, onLaunch);
+
+            ToastNotification toast = new ToastNotification(toastContent.GetXml());
+
+            toast.ExpirationTime = DateTime.Now.AddDays(2);
+            toast.Group = "onInstall";
+
+            ToastNotificationManager.CreateToastNotifier().Show(toast);
+        }
+
+        public static void AppUpdatedToast()
+        {
+            string title = "Power Toys Updated";
+            string content = "";
+            string logo = "../Assets/MiniLogo.png";
+
+            ToastVisual visual = createToastVisual(title, content, logo);
+
+            string onActionClick = new QueryString()
+                {
+                    { "action", "openApp"},
+                    { "status", "openUpdate"},
+                }.ToString();
+
+            ToastActionsCustom actions = createToastAction("See Updates", onActionClick);
+
+            string onLaunch = new QueryString()
+                {
+                    {"action", "openApp"},
+                }.ToString();
+
+            ToastContent toastContent = createToastContent(visual, actions, onLaunch);
+
+            ToastNotification toast = new ToastNotification(toastContent.GetXml());
+
+            toast.ExpirationTime = DateTime.Now.AddDays(2);
+            toast.Group = "update";
+
+            ToastNotificationManager.CreateToastNotifier().Show(toast);
+        }
+
+        public static void AppNeedsUpdateToast()
+        {
+            string title = "PowerToys Update Available";
+            string content = "";
+            string logo = "../Assets/MiniLogo.png";
+
+            ToastVisual visual = createToastVisual(title, content, logo);
+
+            string onActionClick = new QueryString()
+                {
+                    // Action for updating app
+                }.ToString();
+
+            ToastActionsCustom actions = createToastAction("Update", onActionClick);
+
+            string onLaunch = new QueryString()
+                {
+                    {"action", "openApp"},
+                }.ToString();
+
+            ToastContent toastContent = createToastContent(visual, actions, onLaunch);
+
+            ToastNotification toast = new ToastNotification(toastContent.GetXml());
+
+            toast.ExpirationTime = DateTime.Now.AddDays(2);
+            toast.Group = "updateAvailable";
+
+            ToastNotificationManager.CreateToastNotifier().Show(toast);
```

Comment:
Does this prevent against duplicate notifications of the same type?

Reply:
Each notification should only show up once per event (install and update) so if  someone did not open the toast from the last update, we still want them to  receive the next one

Reply:
Would 2 app updates cause two notifications that look the same?